### PR TITLE
show_subject_for_user

### PR DESCRIPTION
**Pull request by hienbx94**
I have created a pull request. Please check it.

### CODE REVIEW

**Comment by ttuan** [comment]
OK

File: app/controllers/user_subjects_controller.rb [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+  def show
+    @user_subject.subject.tasks.each do |task|
+      @user_subject.user_tasks.find_or_initialize_by task_id: task.id,
+        user_id: current_user.id
+    end
+  end
+
+  private
+  def init_course
+    @user_subject = UserSubject.find params[:id]
```
Comment (VuongLong) [inline app/controllers/user_subjects_controller.rb]: load_and_authorize_resource

File: app/controllers/user_subjects_controller.rb [hunk 2]
```diff
@@ -0,0 +1,16 @@
+class UserSubjectsController < ApplicationController
+  load_and_authorize_resource :user_subject
```
Comment (VuongLong) [inline app/controllers/user_subjects_controller.rb]: ko cần user_subject đâu anh ^^

**Comment by VuongLong** [comment]
ok

**Comment by lannp** [comment]
OK

File: app/controllers/user_subjects_controller.rb [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+  def show
+    @user_subject.subject.tasks.each do |task|
+      @user_subject.user_tasks.find_or_initialize_by task_id: task.id,
+        user_id: current_user.id
+    end
+  end
+
+  private
+  def init_course
```
Comment (dieunb) [inline app/controllers/user_subjects_controller.rb]: `load_course`

**Pull request merged**